app/controllers/rooms_controller.rb

```diff
@@ -4,7 +4,7 @@ class RoomsController < ApplicationController
   before_action :set_room, only: %i[show update]
   # GET /rooms/:room_hash
   def show
-    @jams = @room.jams.order('jams.created_at DESC').to_a
+    @jams = @room.jams.limit(20).order('jams.created_at DESC').to_a
```

Comment:
`@room.jams.limit(20)` should be enough, you don't need `.order('jams.created_at DESC').to_a` anymore